gradio/routes.py

```diff
@@ -428,12 +428,18 @@ async def file(path_or_url: str, request: fastapi.Request):
                 return RedirectResponse(
                     url=path_or_url, status_code=status.HTTP_302_FOUND
                 )
+
+            invalid_prefixes = ["//", "file://", "ftp://", "sftp://", "smb://"]
```

Comment:
I believe checking whether or not `abs_path` exists will also fix the issue and it doesn't involve hardcoding this list of prefixes?

Reply:
So I'm not sure totally sure but it seems like any file operation on Windows could leak credentials

Reply:
Yikes

Reply:
Do we care about ipfs urls here too?

Reply:
Probably so I don't know that much about them, but let me rewrite them to prevent any protocols except for http/https